src/datatypes.rs

```diff
@@ -331,10 +353,20 @@ fn rect_data_type(dim: Dimension) -> DataType {
 }
 
 impl GeoDataType {
-    /// Convert a [`GeoDataType`] into the relevant arrow [`DataType`].
+    /// Converts a [`GeoDataType`] into the relevant arrow [`DataType`].
     ///
     /// Note that an arrow [`DataType`] will lose the accompanying GeoArrow metadata if it is not
     /// part of a [`Field`] with GeoArrow extension metadata in its field metadata.
+    ///
+    /// # Examples
+    ///
+    /// ```
+    /// use geoarrow::{array::CoordType, datatypes::{GeoDataType, Dimension}};
```

Comment:
As a general question: how do you write these docstrings? It's quite annoying to write inside a markdown comment; do you write them separately with rust-analyzer for help and then copy them in here?

Reply:
I don't, I usually write them by hand in the docstring block. It may be a bit of a hairshirt, but my personal approach is that if I can't easily write the usage docs kinda from scratch, then my interface might be too clumsy.

Reply:
I can barely write rust at all without rust-analyzer 😅